GitCommands/Settings/Setting.cs

```diff
@@ -8,14 +10,29 @@ public abstract class Setting<T>
         public readonly T DefaultValue;
         public readonly string Name;
 
+        public event Action Updated;
+
         protected Setting(string name, SettingsPath settingsSource, T defaultValue)
         {
             Name = name;
             SettingsSource = settingsSource;
             DefaultValue = defaultValue;
         }
 
-        public abstract T Value { get; set; }
+        public virtual T Value
+        {
+            get => GetVal(Name, DefaultValue);
+            set
+            {
+                if (Value.Equals(value))
+                {
+                    return;
+                }
+
+                SetVal(Name, value);
+                Updated?.Invoke();
```

Comment:
Can we use [`INotifyPropertyChanged `](https://docs.microsoft.com/dotnet/api/system.componentmodel.inotifypropertychanged?view=netframework-4.6.1) instead?

Reply:
We can. But I think this is a little for another case and we have one property.